### PR TITLE
Added Adaptive Pooling

### PR DESCRIPTION
**Pull request by dnabanita7**
I have added ``AdaptiveMaxPool`` and ``AdaptiveMeanPool`` so that we can do a similar [PyTorch implementation](https://github.com/darsnack/FluxModels.jl/issues/1#issue-635582192). cc @darsnack 

### PR 

- [x] Tests are added
- [ ] Entry in NEWS.md
- [x] Documentation, if applicable
- [ ] Final review from `@MikeInnes` or `@dhairyagandhi96` (for API changes).

### Flux issue linking
[Flux#1224](https://github.com/FluxML/Flux.jl/issues/1224)
### MLH issue linking
[0.3.x-projects#26](https://github.com/MLH-Fellowship/0.3.x-projects/issues/26)

### CODE REVIEW

File: src/layers/conv.jl [hunk 1]
```diff
@@ -480,6 +480,71 @@ end
...
+Adaptive Max pooling layer. `out` is the size of the dimension of the output.
+
+"""
+struct AdaptiveMaxPool{O}
+    out::NTuple{O,Int}
+end
+
+function AdaptiveMaxPool(out::NTuple{O,Int})
+  return AdaptiveMaxPool(out)
+end
```
Comment (darsnack) [inline src/layers/conv.jl]: This constructor is created by default when you define the struct. So no need to write this.

File: src/layers/conv.jl [hunk 2]
```diff
@@ -480,6 +480,71 @@ end
...
+Adaptive Mean pooling layer. `out` is the size of the dimension of the output.
+
+"""
+struct AdaptiveMeanPool{O}
+    out::NTuple{O,Int}
+end
+
+function AdaptiveMeanPool(out::NTuple{O,Int})
+  return AdaptiveMeanPool(out)
+end
```
Comment (darsnack) [inline src/layers/conv.jl]: Same as above.

**Review by darsnack — CHANGES_REQUESTED** [review]
This looks good, but we'll need to add tests and docs in order to merge this in Flux#master. I reviewed the changes, but we'll probably want a final review from @dhairyagandhi96 once this is ready.

**Comment by darsnack** [comment]
Based on https://github.com/MLH-Fellowship/FluxModels.jl/pull/1#issuecomment-644842859, I would close this PR and commit these changes directly to the MLH-Fellowship/Flux.jl repo. Then open a PR on FluxML/Flux.jl for review.

File: src/layers/conv.jl (hunk 1 above)
Comment (dnabanita7) [reply src/layers/conv.jl]: I forgot to delete it.

**Review by darsnack — CHANGES_REQUESTED** [review]
There is an extra `conv.jl.save` file added? Could you delete that?

File: src/layers/conv.jl [hunk 3]
```diff
@@ -480,6 +480,63 @@ end
 outdims(l::CrossCor, isize) =
   output_size(DenseConvDims(_paddims(isize, size(l.weight)), size(l.weight); stride = l.stride, padding = l.pad, dilation = l.dilation))
 
+"""
+    AdaptiveMaxPool(out)
+
+Adaptive Max pooling layer. `out` is the size of the dimension of the output.
```
Comment (darsnack) [inline src/layers/conv.jl]: Minor issues: Change "Adaptive Max pooling layer" to "Adaptive max pooling layer."

Can you add a sentence mentioning that the output size is fixed for arbitrary input sizes?

File: src/layers/conv.jl [hunk 4]
```diff
@@ -480,6 +480,63 @@ end
...
+end
+
+function Base.show(io::IO, a::AdaptiveMaxPool)
+  print(io, "AdaptiveMaxPool(", a.out, ")")
+end
+
+"""
+    AdaptiveMeanPool(out)
+
+Adaptive Mean pooling layer. `out` is the size of the dimension of the output.
```
Comment (darsnack) [inline src/layers/conv.jl]: Same comments as max pool.

**Review by darsnack — CHANGES_REQUESTED** [review]
You also need to add the docstrings to `docs/src/models/layers.md`.

**Review by darsnack — APPROVED** [review]
Awesome. You can merge this yourself then make a PR to FluxML/Flux.jl.

**Pull request merged**